### ISSUE TITLE
Move Dreditor out of Drupal namespace and create extensions event handlers

### ISSUE THREAD

**Issue by markhalliwell** [unicorn-fail/dreditor#155]
Currently we have everything in `Drupal.dreditor`. Before we ~~can~~ do #106, we should ~~really need~~ probably to move into our own `Dreditor` namespace.

**Comment by sun** [comment]
I agree with

``` diff
-Drupal.dreditor.*
+Dreditor.*
```

However, let's bear in mind that we're injecting our behaviors into `Drupal.behaviors.*` of drupal.org pages, so whenever drupal.org's native JS calls `Drupal.attachBehaviors()`, Dreditor's behaviors are participating.  Changing that would be non-trivial.

**Comment by markhalliwell** [comment]
Yes. Personally I'd rather abandon the whole "behaviors" name entirely in Dreditor (makes it confusing at times).

Instead I think we should have plugin `init` and `bind` methods that get called on a page. Not entirely sure how we want to go about it, but it will likely need to happen before this one can. I'll create another issue.

**Comment by markhalliwell** [comment]
I'm increasing the scope of this issue to also include adding native extension event handling because we'll need to process d.o's native Drupal.attachBevahior events and pass them to Dreditor.

**Comment by markhalliwell** [comment]
Dreditor is being decommissioned due to the lack of availability of current maintainers and the fact that Drupal.org is becoming more feature rich at a much faster rate than Dreditor itself now.

This decision was also partially made due to the amount of work (complete rewrite) it would take to create an "official" Firefox extension (#256).

If you already have the extension installed, it should continue working for a while until, inevitably,  the markup on Drupal.org changes. It will no longer be updated.

We understand that many of you love this extension, as do we. This isn't the end, in fact, it's a much brighter beginning. All the things we love about Dreditor can, and should, be moved into Drupal.org natively.

For more information, please follow: https://www.drupal.org/node/2779729

**Comment by subhojit777** [comment]
This issue was moved to dreditor/dreditor#28
